scripts/ipintutil

```diff
@@ -0,0 +1,265 @@
+                admin = get_if_admin_state(iface, namespace)
+                oper = get_if_oper_state(iface, namespace)
+                master = get_if_master(iface)
+
+            ip_intf_attr = {"vrf": master,
+                              "ipaddr": natsorted(ifaddresses),
+                              "admin": admin,
+                              "oper": oper,
+                              "bgp_neighs": bgp_neighs,
+                              "ns": namespace}
+
+            ip_intfs[iface] = ip_intf_attr
+    return ip_intfs
+
+
+def display_ip_intfs(ip_intfs):
+    header = ['Interface', 'Master', 'IPv4 address/mask',
+              'Admin/Oper', 'BGP Neighbor', 'Neighbor IP']
+    data = []
+    for ip_intf, v in natsorted(ip_intfs.items()):
+        ip_address = v['ipaddr'][0][1]
+        neighbour_name = v['bgp_neighs'][ip_address][0]
+        neighbour_ip = v['bgp_neighs'][ip_address][1]
+        data.append([ip_intf, v['vrf'], v['ipaddr'][0][1], v['admin'] + "/" + v['oper'],  neighbour_name,  neighbour_ip])
+        for ifaddr in v['ipaddr'][1:]:
+            neighbour_name = v['bgp_neighs'][ifaddr[1]][0]
+            neighbour_ip = v['bgp_neighs'][ifaddr[1]][1]
+            data.append(["", "", ifaddr[1], "", neighbour_name, neighbour_ip])
+    print(tabulate(data, header, tablefmt="simple", stralign='left', missingval=""))
+
+
+def get_ip_intfs(af, namespace, display):
+    '''
+    Get all the ip interface present on the device.
+    This include ip interfaces on the host as well as ip
+    interfaces in each network namespace
+    '''
+    device = multi_asic_util.MultiAsic(namespace_option=namespace,
+                                       display_option=display)
+    namespace_list = device.get_ns_list_based_on_options()
+
+    # for single asic devices there is one namespace DEFAULT_NAMESPACE
+    # for multi asic devices, there is one network namespace
+    # for each asic and one on the host
+    if device.is_multi_asic:
+        namespace_list.append(constants.DEFAULT_NAMESPACE)
+
+    ip_intfs = {}
+    for namespace in namespace_list:
+        ip_intfs_in_ns = get_ip_intfs_in_namespace(af, namespace, display)
+        # multi asic device can have same ip interface in different namespace
+        # so remove the duplicates
+        if device.is_multi_asic:
+            for ip_intf, v in ip_intfs_in_ns.items():
+                if ip_intf in ip_intfs:
+                    if v['ipaddr'] != ip_intfs[ip_intf]['ipaddr']:
+                        ip_intfs[ip_intf]['ipaddr'] += (v['ipaddr'])
+                        ip_intfs[ip_intf]['bgp_neighs'].update(v['bgp_neighs'])
+                    continue
+                else:
+                    ip_intfs[ip_intf] = v
+        else:
+            ip_intfs.update(ip_intfs_in_ns)
+    return ip_intfs
+
+
+def main():
+    # This script gets the ip interfaces from different linux
+    # network namespaces. This can be only done from root user.
+    cmd_exec_in_root()
```

Comment:
The interface information in the kernel in different namespace, cannot be fetched with regular user so the script has be executed as   `root`

Reply:
Function name is a bit unclear. Function is only two lines and appears to be only used once, here in `main`. Doe it even need to be a function?

Reply:
Removed the function.